crates/l3/common/src/lib.rs

```diff
@@ -0,0 +1,113 @@
+pub mod errors;
+use color_eyre::{eyre::eyre, Result};
+use starknet_accounts::{Account, Call, ConnectedAccount, Execution, SingleOwnerAccount};
+use starknet_contract::ContractFactory;
+use starknet_core::types::contract::{CompiledClass, SierraClass};
+use starknet_core::types::{BlockId, BlockTag, FunctionCall, InvokeTransactionResult};
+use starknet_core::utils::get_selector_from_name;
+use starknet_ff::FieldElement;
+use starknet_providers::jsonrpc::{HttpTransport, JsonRpcClient};
+use starknet_providers::Provider;
+use starknet_signers::LocalWallet;
+use std::sync::Arc;
+
+pub type LocalWalletSignerMiddleware =
+    SingleOwnerAccount<JsonRpcClient<HttpTransport>, LocalWallet>;
+
+type RpcAccount<'a> = SingleOwnerAccount<&'a JsonRpcClient<HttpTransport>, LocalWallet>;
+pub type TransactionExecution<'a> = Execution<'a, RpcAccount<'a>>;
+
+pub const NO_CONSTRUCTOR_ARG: Vec<FieldElement> = Vec::new();
+
+pub const MAX_FEE: &str = "0x1000000000000";
+
+pub trait StarknetContractClient {
+    fn address(&self) -> FieldElement;
+    fn client(&self) -> LocalWalletSignerMiddleware;
+}
+
+pub async fn invoke_contract(
+    client: &Arc<&LocalWalletSignerMiddleware>,
```

Comment:
It looks like you are only using `client` to do `client.as_ref()` so can you change the signature so that the type of `client` is whatever `client.as_ref()` returns?

Reply:
updated to return &LocalWalletSignerMiddleware as client.as_ref() returns a reference to it.